### ISSUE TITLE
Use alias names in error messages

### ISSUE THREAD

**Issue by rhettinger** [python/mypy#2968]
Given:

```
from typing import Tuple
from math import hypot

Point = Tuple[float, ...]
Centroid = Point

def dist(p: Centroid, q: Point) -> float:
    px, py = p
    qx, qy = q
    return hypot(px - qx, py - qy)

print( dist('xyz', (3.5, 7.2)) )
```

*mypy* detect the incorrect first argument:

```
$ mypy tmp.py
tmp.py:12: error: Argument 1 to "dist" has incompatible type "str"; expected Tuple[float, ...]
```

However, it would be much nicer is the error reporting could have used the alias name:

`tmp.py:12: error: Argument 1 to "dist" has incompatible type "str"; expected "Centroid", which expands to Tuple[float, ...]
`

I realize it may be easy to do, but it would improve the usability of aliases.

**Comment by JukkaL** [comment]
Agreed that your proposed message would be better. Mypy currently expands aliases eagerly and doesn't keep track of the name of the alias, so this would be somewhat non-trivial to support.

**Comment by JelleZijlstra** [comment]
I think some work has been done since 2017 that should make it a bit easier. Look for `TypeAliasType` in the code.

**Comment by pranavrajpal** [comment]
I've looked into this, and there are 2 issues that will probably make this somewhat hard to implement completely:
1. There are a decent number of places where code calls `get_proper_type` on arguments and then passes on the expanded alias to another function, instead of the original alias, despite `get_proper_type`'s docstring saying not to do that. That means that the error handling code sometimes only sees the expanded alias, so it doesn't have enough info to show the alias name in the error.
2. There's currently an optimization where the types for type aliases are eagerly expanded for aliases to types that are relatively simple (see [here](https://github.com/python/mypy/blob/f19a711eae747e2e5525bf0b0baaa3d3ae972fbd/mypy/semanal.py#L2748) for more details). That means that, again, the alias name isn't available to the error message printing code, so we're probably going to have to remove that optimization.

I should be able to create a PR soon that starts to implement this. So far, I'm just fixing examples of problem 1 everywhere I happen to find, and I'm leaving out a fix for problem 2 (removing that optimization doesn't seem hard but it definitely seems nontrivial).
